tutorials/dataframe/df107_SingleTopAnalysis.py

```diff
@@ -0,0 +1,245 @@
+## \file
+## \ingroup tutorial_dataframe
+## \notebook -draw
+## \brief A single top analysis using the ATLAS Open Data release of 2020, with RDataFrame.
+##
+## This tutorial is the analysis of single top production adapted from the ATLAS Open Data release in 2020
+## (http://opendata.atlas.cern/release/2020/documentation/). The data was recorded with the ATLAS detector
+## during 2016 at a center-of-mass energy of 13 TeV. Top quarks with a mass of about 172 GeV are preliminary
+## produced in pairs but also appear alone, dominantly from the decays of a W boson in association with a light jet.
+##
+## The analysis is translated to a RDataFrame workflow processing a subset of 60 GB of simulated events and data.
+## The original analysis is reduced by a factor given as the command line argument --lumi-scale.
+##
+## \macro_image
+## \macro_code
+## \macro_output
+##
+## \date July 2020
+## \author Stefan Wunsch (KIT, CERN)
+
+import ROOT
+import json
+import argparse
+import os
+
+# Argument parsing
+parser = argparse.ArgumentParser()
+parser.add_argument("--lumi-scale", default=0.05, help="Scale down from the overall available dataset of 10 fb^-1")
+parser.add_argument("-b", action="store_true", default=False, help="Use ROOT batch mode")
+parser.add_argument("-t", action="store_true", default=False, help="Use implicit multi threading (running always on the full dataset)")
+args = parser.parse_args()
+if args.b: ROOT.gROOT.SetBatch(True)
+if args.t:
+    ROOT.EnableImplicitMT()
+    lumi_scale = 1.0
+else:
+    lumi_scale = args.lumi_scale
+
+# Create a ROOT dataframe for each dataset
+# Note that we load the filenames from an external json file.
+path = "root://eospublic.cern.ch//eos/opendata/atlas/OutreachDatasets/2020-01-22"
+files = json.load(open(os.path.join(os.environ["ROOTSYS"], "tutorials/dataframe", "df107_SingleTopAnalysis.json")))
+processes = files.keys()
+df = {}
+xsecs = {}
+sumws = {}
+samples = []
+for p in processes:
+    for d in files[p]:
+        # Construct the dataframes
+        folder = d[0] # Folder name
+        sample = d[1] # Sample name
+        xsecs[sample] = d[2] # Cross-section
+        sumws[sample] = d[3] # Sum of weights
+        num_events = d[4] # Number of events
+        samples.append(sample)
+        df[sample] = ROOT.RDataFrame("mini", "{}/1lep/{}/{}.1lep.root".format(path, folder, sample))
+
+        # Scale down the datasets if requested
+        if not (args.t or lumi_scale == 1.0):
+            df[sample] = df[sample].Range(int(num_events * lumi_scale))
+
+# Select events for the analysis and make histograms of the top mass
+ROOT.gInterpreter.Declare("""
```

Comment:
I would combine the two `Declare` calls together and add a comment like "This code gets declared to the ROOT C++ interpreter" since this is an educational example.

Reply:
You mean this declare and second one for the compuation of the top mass? I like that the impl is close the the location where it's used!

Reply:
But I fully agree that I could make more clear what happens in a `Declare` call!